dummy_app.py

```diff
@@ -34,9 +36,30 @@
     st.pyplot(fig)
 
 # load library file in sidebar
+library_spectrums = []  # default so later code doesn't crash
 library_file = st.sidebar.file_uploader("Choose a spectra library file...",
                                         type=['json', 'txt'])
 if library_file is not None:
     if library_file.name.endswith("json"):
         library_file.seek(0)  # fix for streamlit issue #2235
         library_spectrums = json_loader(library_file)
+
+# processing of query and library spectra
+st.write("""## Post-process spectra
+Spec2Vec similarity scores rely on creating a document vector for each
+spectrum. For the underlying word2vec model we want the documents (=spectra) to
+be more homogeneous in their number of unique words. Assuming that larger
+compounds will on average break down into a higher number of meaningful
+fragment peaks we reduce the document size of each spectrum according to its
+parent mass.
+""")
+# todo: we could add some buttons later to make this adjustable
+with st.beta_expander("View processing defaults"):
+    st.markdown("""* normalize peaks (maximum intensity to 1)\n* remove peaks 
```

Comment:
That looks very nice in the app, so nice addition!
However, I think it would be better to read this from the actual defaults that we define in the function. 
Probably takes some fiddling, but otherwise it is usually just a matter of time before one get's changed while forgetting to also change the other one...

Reply:
Maybe that would also be a good point to why to redefine the parameter setting thing. You could add an additional function like
```python
def set_spec2vec_defaults(**settings):
    """Set spec2vec default argument values"(where no user input is given)"."""
    defaults = {...
    }

    # Set default parameters or replace by **settings input
    for key in defaults:
        if key in settings:
            print(f"The value of {key} is set from {defaults[key]} (default) to {settings[key]}")
        else:
            settings[key] = defaults[key]
    return settings
```
And then also read the default setting from that for displaying?

Reply:
Ah yes that is a good solution actually. Thanks!